### ISSUE TITLE
GPU available but not used

### ISSUE THREAD

**Issue by alevangel** [openvinotoolkit/anomalib#232]
Executing CFLOW model on Colab, PyTorch returns this

```
GPU available: True, used: False
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs

GPU available but not used. Set "accelerator" and "devices" using Trainer(accelerator='gpu', devices=1).
```

Even if i'm setting gpus=1 on the config.yml of cflow.

 
**To Reproduce**
`train.py --model cflow
`

**GPU Model**
GPU 0: Tesla T4

**Versions**
```
pytorch-lightning             1.6.1
torch                         1.10.0+cu111
torchaudio                    0.10.0+cu111
torchmetrics                  0.7.3
torchsummary                  1.5.1
torchtext                     0.11.0
torchvision                   0.11.1+cu111
```

**Comment by samet-akcay** [comment]
Hi @alevangel, Which GPU model do you have?

**Comment by alevangel** [comment]
> Hi @alevangel, Which GPU model do you have on Colab?

GPU 0: Tesla T4 (edited)

**Comment by ashwinvaidya17** [comment]
@alevangel Can you share your torch version?

**Comment by alevangel** [comment]
> @alevangel Can you share your torch version?
```
pytorch-lightning             1.6.1
torch                         1.10.0+cu111
torchaudio                    0.10.0+cu111
torchmetrics                  0.7.3
torchsummary                  1.5.1
torchtext                     0.11.0
torchvision                   0.11.1+cu111
```

**Comment by alevangel** [comment]
I solved rolling back at version of pytorch-lightning==1.5.9

**Comment by andriy-onufriyenko** [comment]
I have the same problem with CFLOW model on my local machine.
```
GPU available: True, used: False
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs

GPU available but not used. Set `accelerator` and `devices` using `Trainer(accelerator='gpu', devices=1)
```

`python tools/train.py --model cflow
`

GPU:
NVIDIA GeForce GTX 1650 Ti

`pytorch-lightning             1.6.1`

**Comment by alevangel** [comment]
@andriy-onufriyenko use pytorch-lightning==1.5.9

**Comment by andriy-onufriyenko** [comment]
> @andriy-onufriyenko use pytorch-lightning==1.5.9

Install pytorch-lightning==1.5.9 and have another problem now:

`ModuleNotFoundError: No module named 'torchtext.legacy'`

**Comment by samet-akcay** [comment]
@andriy-onufriyenko, can you add `torch>=1.8.1` above [this](https://github.com/openvinotoolkit/anomalib/blob/development/requirements/base.txt#L8) line, and do `pip install -e` again?

**Comment by andriy-onufriyenko** [comment]
> 

@samet-akcay 

```
albumentations>=1.1.0
einops>=0.3.2
kornia>=0.5.6
omegaconf>=2.1.1
opencv-python>=4.5.3.56
pandas>=1.1.0
pytorch-lightning>=1.6.0
torch>=1.8.1
torchvision>=0.9.1
torchtext>=0.9.1
wandb==0.12.9
matplotlib>=3.4.3
```
It's working but without GPU

```
GPU available: True, used: False
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs

GPU available but not used. Set `accelerator` and `devices` using `Trainer(accelerator='gpu', devices=1)`
```
Trying edit `config.yaml` :
```
# PL Trainer Args. Don't add extra parameter here.
trainer:
  accelerator: 'gpu'
```
and got

`ValueError: Unsupported accelerator found: gpu. Should be one of [null, ddp]`

**Comment by samet-akcay** [comment]
> I solved rolling back at version of pytorch-lightning==1.5.9

I found the problem. PyTorch Lightning deprecated and / or modified some of the `Trainer` configs after `v1.6.0`, which is the reason for not using the GPU despite being available.

The issue has been fixed in PR #234

**Comment by andriy-onufriyenko** [comment]
> > I solved rolling back at version of pytorch-lightning==1.5.9
> 
> I found the problem. PyTorch Lightning deprecated and / or modified some of the `Trainer` configs after `v1.6.0`, which is the reason for not using the GPU despite being available.
> 
> The issue has been fixed in PR #234

@samet-akcay

I checked it works for me:

```
GPU available: True, used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs

```
Just replace all 11 files from PR [234](https://github.com/openvinotoolkit/anomalib/pull/234) in to my project.
